dummyserver/testcase.py

```diff
@@ -181,19 +181,28 @@ class HTTPDummyProxyTestCase:
     http_host_alt = "127.0.0.1"
     http_server: ClassVar[httpserver.HTTPServer]
     http_port: ClassVar[int]
+    http_url: ClassVar[str]
+    http_url_alt: ClassVar[str]
 
     https_host = "localhost"
     https_host_alt = "127.0.0.1"
     https_certs = DEFAULT_CERTS
     https_server: ClassVar[httpserver.HTTPServer]
     https_port: ClassVar[int]
+    https_url: ClassVar[str]
+    https_url_alt: ClassVar[str]
 
     proxy_host = "localhost"
     proxy_host_alt = "127.0.0.1"
     proxy_server: ClassVar[httpserver.HTTPServer]
     proxy_port: ClassVar[int]
+    proxy_url: ClassVar[str]
     https_proxy_server: ClassVar[httpserver.HTTPServer]
     https_proxy_port: ClassVar[int]
+    https_proxy_url: ClassVar[str]
+
+    certs_dir = ""
```

Comment:
Should these be `ClassVar[str]` as well?

Reply:
as we don't have `ClassVar` for the other class variables with default values in this file, I didn't define it by `ClassVar`
for example:
https://github.com/urllib3/urllib3/blob/7c9ca88b7dd5fe6b0e9fac52f72202edd74bd8b2/dummyserver/testcase.py#L191

Should we change them as well?

Reply:
It looks like they get set on the `cls` so they should be `ClassVar` as well. If `ClassVar` is not used then it is taken to be a default for instance variables.

Reply:
Ok, Thanks for the explanation. Changed them to `ClassVar`